lms/templates/discussion/_filter_dropdown.html

```diff
@@ -6,8 +6,8 @@
 %>
 
 <%def name="render_dropdown(map)">
-    % for child in map["children"]:
-        % if child in map["entries"]:
+    % for child, c_type in map["children"]:
+        % if child in map["entries"] and c_type == 'entry':
```

Comment:
Thanks for tracking this down @attiyaIshaque.  I imagine this is the main purpose of adding the `type`.

Are there any uses of this `type` in the JavaScript?  If so, could you point it out?  The reason I ask, is because:
a) The array of tuples in Python gives a slightly more awkward JavaScript data-structure of array of arrays.
b) It may greatly simplify this PR to not have to touch the JavaScript if there is a simple way to separate the array required by Python from the data sent to JavaScript.

Also, was the only problem one of entry vs subcategory?  If topic names are repeated at multiple levels of subcategories, will we get a similar type problem, or is this ok?